src/main/java/org/spongepowered/common/command/MinecraftCommandWrapper.java

```diff
@@ -133,41 +135,50 @@ public PluginContainer getOwner() {
 
     @Override
     public boolean testPermission(CommandSource source) {
-        if (source instanceof ICommandSender) {
-            return this.command.canCommandSenderUseCommand((ICommandSender) source);
-        } else {
-            return source.hasPermission(getCommandPermission());
-        }
+        return this.command.canCommandSenderUseCommand(WrapperICommandSender.of(source));
     }
 
     @Override
     public Optional<Text> getShortDescription(CommandSource source) {
-        return Optional.absent();
+        return getHelp(source);
     }
 
     @Override
     public Optional<Text> getHelp(CommandSource source) {
-        return Optional.absent();
+        String translation = command.getCommandUsage(WrapperICommandSender.of(source));
+        if (translation == null) {
+            return Optional.absent();
+        }
+        return Optional.of((Text) Texts.of(new SpongeTranslation(translation)));
     }
 
     @Override
     public Text getUsage(CommandSource source) {
-        final ICommandSender mcSender =
-                source instanceof ICommandSender ? (ICommandSender) source : new WrapperICommandSender(source);
+        final ICommandSender mcSender = WrapperICommandSender.of(source);
         String usage = this.command.getCommandUsage(mcSender);
-        if (usage.startsWith("/") && usage.contains(" ")) {
-            usage = usage.substring(usage.indexOf(" "));
+        Translation translation = Sponge.getGame().getRegistry().getTranslationById(usage).get();
+        if (source instanceof Player) {
+            usage = translation.get(((Player) source).getLocale());
```

Comment:
Yeah -- I think the best solution to this might just be to add a getLocale method to CommandSource -- it's something I've wanted to do for a while and the idea keeps sounding better.